scene/3d/xr_nodes.cpp

```diff
@@ -382,6 +383,21 @@ void XRNode3D::_removed_tracker(const StringName p_tracker_name, int p_tracker_t
 void XRNode3D::_pose_changed(const Ref<XRPose> &p_pose) {
 	if (p_pose.is_valid() && p_pose->get_name() == pose_name) {
 		set_transform(p_pose->get_adjusted_transform());
+		_set_has_tracking_data(p_pose->get_has_tracking_data());
+	}
+}
+
+void XRNode3D::_pose_lost_tracking(const Ref<XRPose> &p_pose) {
+	if (p_pose.is_valid() && p_pose->get_name() == pose_name) {
+		_set_has_tracking_data(false);
+	}
+}
+
+void XRNode3D::_set_has_tracking_data(bool p_has_tracking_data) {
+	// Handle change of has_tracking_data
+	if (p_has_tracking_data != has_tracking_data) {
+		has_tracking_data = p_has_tracking_data;
+		emit_signal(SNAME("tracking_changed"), has_tracking_data);
 	}
```

Comment:
```suggestion
	if (p_has_tracking_data == has_tracking_data) {
		return;
	}

	has_tracking_data = p_has_tracking_data;
	emit_signal(SNAME("tracking_changed"), has_tracking_data);
```
We normally do an early exit and put the active code below, it makes it easier to read IMO

Reply:
Modified to use early-exit style.